File: baselibrary/src/main/java/com/ju/baselibrary/widget/gloading/GlobalAdapter.java
package com.ju.baselibrary.widget.gloading;

import android.view.View;

import com.ju.baselibrary.utils.Constants;

public class GlobalAdapter implements Gloading.Adapter {

    @Override
    public View getView(Gloading.Holder holder, View convertView, int status) {
        GlobalLoadingStatusView loadingStatusView = null;
        //reuse the old view, if possible
        if (convertView != null && convertView instanceof GlobalLoadingStatusView) {
            loadingStatusView = (GlobalLoadingStatusView) convertView;
        }
        if (loadingStatusView == null) {
            loadingStatusView = new GlobalLoadingStatusView(holder.getContext(), holder.getRetryListener());
        }
        loadingStatusView.setStatus(status);
        Object data = holder.getData();
        //show or not show msg view
        boolean hideMsgView = Constants.HIDE_LOADING_STATUS_MSG.equals(data);
        loadingStatusView.setMsgViewVisibility(!hideMsgView);
        return loadingStatusView;
    }
}

File: baselib/src/main/java/com/hzy/baselib/widget/gloading/GlobalAdapter.kt
package com.hzy.baselib.widget.gloading

import android.view.View

import com.hzy.baselib.util.Constants


class GlobalAdapter : Gloading.Adapter {

    override fun getView(holder: Gloading.Holder, convertView: View?, status: Int): View {
        var loadingStatusView: GlobalLoadingStatusView? = null
        //reuse the old view, if possible
        if (convertView != null && convertView is GlobalLoadingStatusView) {
            loadingStatusView = convertView
        }
        if (loadingStatusView == null) {
            loadingStatusView = GlobalLoadingStatusView(holder.context, holder.retryListener)
        }
        loadingStatusView.setStatus(status)
        val data = holder.getData<Any>()
        //show or not show msg view
        val hideMsgView = Constants.HIDE_LOADING_STATUS_MSG == data
        loadingStatusView.setMsgViewVisibility(!hideMsgView)
        return loadingStatusView
    }
}
